src/Type/Row/Homogeneous.purs

```diff
@@ -0,0 +1,22 @@
+module Type.Row.Homogeneous
+  ( class FieldOf
+  , class Homogeneous
+  ) where
+
+-- | Ensure that every field in a row has the same type.
+import Type.Equality (class TypeEquals)
+import Type.Row (class RowToList, Cons, Nil, kind RowList)
+
+-- | Ensure that every field in a row has the same type.
+class Homogeneous (row :: # Type) fieldType | row -> fieldType
+instance homogeneous
+  :: ( RowToList row fields
+     , FieldOf fields fieldType )
+  => Homogeneous row fieldType
+
+class FieldOf (rowList :: RowList) fieldType | rowList -> fieldType
```

Comment:
It's not clear to me (without reading the docs quite a bit) which one of these should be used in regular code. I'd suggest naming this one something like `HomogeneousRowList` to just be explicit about it.